### PR TITLE
L2 l3 rebase

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- closes https://github.com/RasaHQ/roadmap/issues/841

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/user-guide/building-assistants.rst [hunk 1]
```diff
@@ -8,8 +8,7 @@ Tutorial: Building Assistants
 .. edit-link::
 
 After following the basics of setting up an assistant in the `Rasa Tutorial <https://rasa.com/docs/rasa/user-guide/rasa-tutorial/>`_, we'll
-now walk through building a basic FAQ chatbot and then build a bot that can handle
-contextual conversations.
+now walk through building a basic FAQ assistant.
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: should this not say the original thing again? Since it's still in the same page. Let's discussing moving them to separate pages after this is merged

File: docs/user-guide/building-assistants.rst [hunk 2]
```diff
@@ -144,6 +146,23 @@ You can now train a first model and test the bot, by running the following comma
...
+    Your input ->  bye!
+    Bye!
+    Your input ->  thanks!
+    No worries!
+    Your input ->  hi!
+    Hi
+    Your input ->  goodbye
+    Bye!
+    Your input ->  thank you
+    No worries!
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: any reason we didn't go for a screenshot here btw?

File: docs/user-guide/building-assistants.rst [hunk 3]
```diff
@@ -268,7 +291,7 @@ Now that we’ve defined the NLU side, we need to make Core aware of these chang
      - faq
 
 We’ll also need to add a `retrieval action <https://rasa.com/docs/rasa/core/retrieval-actions/>`_,
-which takes care of sending the response predicted from the ResponseSelector back to the user,
+which takes care of sending the response predicted from the  ResponseSelector back to the user,
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]:
```suggestion
which takes care of sending the response predicted from the ResponseSelector back to the user,
```

File: docs/user-guide/building-assistants.rst [hunk 4]
```diff
@@ -313,22 +336,34 @@ You can read more in this `blog post <https://blog.rasa.com/response-retrieval-m
 `Retrieval Actions <https://rasa.com/docs/rasa/core/retrieval-actions/>`_ page.
 
 Using the features we described in this tutorial, you can easily build a context-less assistant.
-When you’re ready to enhance your assistant with context, check out :ref:`build-contextual-assistant`.
+When you’re ready to enhance your assistant with context, check out :ref:`tutorial-contextual-assistants`.
+
+
+.. note::
+    Here's a minimal checklist of what you need to build a basic FAQ assistant:
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]:
```suggestion
    Here's a minimal checklist of what we covered in this tutorial on building a basic FAQ assistant:
```

File: docs/user-guide/building-assistants.rst [hunk 5]
```diff
@@ -313,22 +336,34 @@ You can read more in this `blog post <https://blog.rasa.com/response-retrieval-m
...
+
+      - ``data/nlu.md``: NLU training data for ``faq/`` intents
+      - ``data/responses.md``: responses associated with ``faq/`` intents
+      - ``config.yml``: ``ReponseSelector`` in your NLU pipeline
+      - ``domain.yml``: a retrieval action ``respond_faq`` and intent ``faq``
+      - ``data/stories.md``: a simple story for FAQs
+      - ``test_stories.md``: E2E test stories for your FAQs
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: Can we make this more like "Create NLU training data for faq/ intents in your nlu.md file" - so it's more narrative, rather than just a list of files

File: docs/user-guide/building-assistants.rst [hunk 6]
```diff
@@ -356,6 +393,8 @@ is clear how this information should be collected.
 For this type of use case, we can use Forms and our FormPolicy. The `FormPolicy <https://rasa.com/docs/rasa/core/policies/#form-policy>`_
 works by predicting the form as the next action until all information is gathered from the user.
 
+
+
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]:
```suggestion
```

File: docs/user-guide/building-assistants.rst [hunk 7]
```diff
@@ -548,15 +587,31 @@ The intents and entities will need to be added to your domain as well:
...
+.. cssclass:: longtable
+
++----------------------------------+--------------------------------------+
+|.. code-block:: md                |.. code-block:: md                    |
+|                                  |                                      |
+|   ## sales form                  |     <!--What's happening here-->     |
+|      - sales_form                |     <!--Run the sales_form action--> |
+|      - form{"name": "sales_form"}|     <!--Activate the form-->         |
+|      - form{"name": null}        |     <!--Deactivate the form-->       |
++----------------------------------+--------------------------------------+
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: why do we have both? I feel like just the first one is enough?

File: docs/user-guide/building-assistants.rst [hunk 8]
```diff
@@ -620,12 +675,38 @@ business logic. Read more about this `here <https://rasa.com/docs/rasa/core/form
...
+          - Add all slots required by the form
+          - Add ``utter_ask_{slot}`` responses for all required slots
+          - Add your form action to the ``forms`` section 
+          - Add all intents and entities from your NLU training data
+      - ``data/stories.md``: 
+          - add a story for the form
+      - ``config.yml``: 
+          - Add the ``FormPolicy`` to your policies
+          - Add entity extractors & dimensions (according to which entities will fill your slots) to your pipeline
+      - ``endpoints.yml``: Define the ``action_endpoint``
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: same with this as in the previous version

File: docs/user-guide/building-assistants.rst [hunk 9]
```diff
@@ -620,12 +675,38 @@ business logic. Read more about this `here <https://rasa.com/docs/rasa/core/form
 However, you should make sure not to handle any unhappy paths inside the form. These
 should be handled by writing regular stories, so your model can learn this behaviour.
 
+
+.. note::
+    Here's a minimal checklist of what you need to do to handle business logic using a form action:
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]:
```suggestion
    Here's a minimal checklist of what we covered in this section on handling business logic using a form action:
```

File: docs/user-guide/building-assistants.rst [hunk 10]
```diff
@@ -620,12 +675,38 @@ business logic. Read more about this `here <https://rasa.com/docs/rasa/core/form
...
+
 Handling unexpected user input
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 All expected user inputs should be handled by the form we defined above, i.e. if the
 user provides the information the bot asks for. However, in real situations, the user
-will often behave differently. In this section we’ll go through various forms of
+will often behave differently.
+
+In this section we’ll go through various forms of
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: why did we move that?

File: docs/user-guide/building-assistants.rst (hunk 8 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: did we want checklists like this for all other sections below as well?

File: docs/user-guide/building-assistants.rst [hunk 11]
```diff
@@ -856,7 +968,7 @@ We define some intent mappings to make it more intuitive to the user what an int
 Out of scope intent
 """""""""""""""""""
 
-It is good practice to also handle questions you know your users may ask, but you don’t necessarily have a skill
+It is good practice to also handle questions you know your users may ask, but for which you don’t necessarily have an action
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: hm i feel like i already commented on this? Should be "you don't necessarily have a user goal handled" yet or similar

File: docs/user-guide/building-assistants.rst (hunk 2 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: I wasn't sure if a screenshot of the commandline was appropriate, but can put it in if it would be better

File: docs/user-guide/building-assistants.rst (hunk 1 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: sorry! thought I reverted that.

**Comment by melindaloubser1** [comment]
I think a few things must have gotten lost - urg. Will readdress

File: docs/user-guide/building-assistants.rst (hunk 7 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: that was an experiment I forgot to remove

File: docs/user-guide/building-assistants.rst (hunk 8 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: I'm putting them in - since these sections aren't quite so didactic, it gets a bit wordy. can remove it if its distracting

File: docs/user-guide/building-assistants.rst (hunk 10 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: I moved examples in and out of that section a few times and missed a backspace

File: docs/user-guide/building-assistants.rst (hunk 11 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: right

File: docs/user-guide/building-assistants.rst (hunk 2 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: what do you think? does it look funny?

File: docs/user-guide/building-assistants.rst [hunk 12]
```diff
@@ -548,15 +578,19 @@ The intents and entities will need to be added to your domain as well:
      - use_case
 
 A story for a form is very simple, as all the slot collection form happens inside the form, and
-therefore doesn’t need to be covered in your stories.
+therefore doesn’t need to be covered in your stories. You just need to write stories showing when the form should be activated. For the sales form, add this story
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: sorry this also still has unaddressed changes from before...
```suggestion
therefore doesn’t need to be covered in your stories. You just need to write a single story showing when the form should be activated. For the sales form, add this story
```

Comment (akelad) [reply docs/user-guide/building-assistants.rst]: can you double check that everything has actually gone in?

File: docs/user-guide/building-assistants.rst (hunk 2 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: nah let's just leave it, it's fine!

File: docs/user-guide/building-assistants.rst (hunk 8 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: what do you mean? you're putting them into other sections as well?

File: docs/user-guide/building-assistants.rst (hunk 6 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: what about this?

**Comment by melindaloubser1** [comment]
sorry didn't mean to rerequest right away!

File: docs/user-guide/building-assistants.rst (hunk 8 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: I did in the last commit - they're at the end of the unexpected input/failing gracefully/ML sections

File: docs/user-guide/building-assistants.rst (hunk 5 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: For a checklist doesn't it make sense to have it by file? since the tutorial is narrative already, it might be helpful to have it like this at the end. Also in the other sections, there's multiple changes per file, and it was a way to organise the changes

Comment (akelad) [reply docs/user-guide/building-assistants.rst]: that makes sense yeah, but then maybe we should say "These are the files we modified during this tutorial" or something similar. Rather than saying "these are the steps we did". Then i'm ok with files :D

File: docs/user-guide/building-assistants.rst (hunk 8 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: ah ok i see!

Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: should they stay?

File: docs/user-guide/building-assistants.rst (hunk 5 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: Yeah that also makes sense, will change the wording

File: docs/user-guide/building-assistants.rst [hunk 13]
```diff
@@ -804,6 +889,31 @@ Then you can retrain your bot and test it again:
...
+          - set the type of ``requested_slot`` to categorical 
+              - add all the required slots from the ``sales_form`` as values
+          - add the ``explain`` intent
+          - add responses for contextual question interruptions (``explain``)
+      - ``data/stories.md``: 
+          - remove stories using mapped intents if you have them
+          - add stories with FAQ interruptions in the middle of filling a form
+          - add stories with contextual question interruptions  (``explain``) in the middle of filling a form
+              - specify the value of ``requested_slot`` before the interruption
+              - the response to the interruption should match the value of ``requested_slot``
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: this checklist seems really long and complex... is there any way we could make it a bit more concise?

File: docs/user-guide/building-assistants.rst [hunk 14]
```diff
@@ -903,6 +1012,28 @@ let’s say the user asks "I want to apply for a job at Rasa", we can then reply
...
+      - ``domain.yml``: 
+          - add the ``out_of_scope`` intent
+          - add an ``utter_out_of_scope`` response
+          - add any specific out of scope intents that you want to handle seperately
+          - add responses for any specific out of scope intents that you want to handle seperately
+      - ``actions.py``: 
+          - customise ``ActionDefaultAskAffirmation`` to suggest intents for the user to choose from
+      - ``config.yml``: 
+          - add the TwoStageFallbackPolicy to the ``policies`` section
+          - define the ``nlu_threshold`` for the TwoStageFallbackPolicy
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: can you make sure these checklists are consistently capitalised? Some are lowercase at the start, others uppercase. I already normalised it for the first two.

File: docs/user-guide/building-assistants.rst [hunk 15]
```diff
@@ -903,6 +1012,28 @@ let’s say the user asks "I want to apply for a job at Rasa", we can then reply
...
+      - ``data/nlu.md``: 
+          - add training data for the ``out_of_scope`` intent
+          - add training data for any specific out of scope intents that you want to handle seperately
+      - ``data/stories.md``:
+          - add stories for any specific out of scope intents that you want to handle seperately
+      - ``domain.yml``: 
+          - add the ``out_of_scope`` intent
+          - add an ``utter_out_of_scope`` response
+          - add any specific out of scope intents that you want to handle seperately
+          - add responses for any specific out of scope intents that you want to handle seperately
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: let's condense that into two bullet points

File: docs/user-guide/building-assistants.rst (hunk 14 above)
Comment (akelad) [reply docs/user-guide/building-assistants.rst]: also, is the nlu_threshold step necessary? Do we actually modify it from the default?

File: docs/user-guide/building-assistants.rst [hunk 16]
```diff
@@ -1016,11 +1147,24 @@ in the conversation below (extracted from a conversation on `Rasa X <https://ras
...
+
+      - ``domain.yml``: 
+          - set featurization of slots used by stories to guide the flow of a conversation
+      - ``data/stories.md``: 
+          - write stories using featurized slots to guide the flow of a conversation
+          - add user conversations successfully handled by an ML policy as stories
+      - ``config.yml`` (in the ``policies`` section): 
+          - set ``max_history`` high enough to account for contextual patterns
+          - replace the MemoizationPolicy with the AugmentedMemoizationPolicy
+          - add the TEDPolicy
```
Comment (akelad) [inline docs/user-guide/building-assistants.rst]: not 100% sure it makes sense to have a checklist here as some of this section is a bit more arbitrary... Or what are your thougths on that?

Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: I agree - It's more like a set of recommendations in the section anyway. I'd rather take it out

File: docs/user-guide/building-assistants.rst (hunk 13 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: how about now?

File: docs/user-guide/building-assistants.rst (hunk 14 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: nope

File: docs/user-guide/building-assistants.rst (hunk 13 above)
Comment (melindaloubser1) [reply docs/user-guide/building-assistants.rst]: how about now?
